sunpy/net/helioviewer.py

```diff
@@ -189,6 +192,63 @@ def download_jp2(self, date, observatory=None, instrument=None, detector=None,
 
         return self._get_file(params, directory=directory, overwrite=overwrite)
 
+    def get_jp2_header(self, date, observatory=None, instrument=None, detector=None,
+                       measurement=None, jp2_id=None, directory=None, overwrite=False):
+        """
+        Get the XML header embedded in a JPEG2000 image. Includes the FITS header as well as a section 
+        of Helioviewer-specific metadata.
+
+        This uses `getJP2Header <https://api.helioviewer.org/docs/v2/#JPEG2000>`_ from the Helioviewer API.
+
+        Parameters
+        ----------
+        observatory : `str`
+            Observatory name
+        instrument : `str`
+            Instrument name
+        measurement : `str`
+            Measurement name
+        detector : `str`
+            Detector name
+        directory : `str`
+            Directory to download JPEG 2000 image to.
+        overwrite : bool
+            Defaults to False.
+            If set to True, will overwrite any files with the same name.
+
+        Returns
+        -------
+        out : `Dict`
+            Returns a dictionary containing the header information of JPEG 2000 image.
+
+        Examples
+        --------
+        >>> from sunpy.net import helioviewer
+        >>> hv = helioviewer.HelioviewerClient()  # doctest: +REMOTE_DATA
+        >>> header = hv.get_jp2_header('2012/07/03', observatory='SDO',
+        ...                            instrument='HMI', detector=None, measurement='continuum')  # doctest: +REMOTE_DATA
+        """
+        if jp2_id is None:
+            try:
+                key = (observatory, instrument, detector, measurement)
+                print(key)
+                source_id = self.data_sources[key]
+                metadata = self.get_closest_image(date, source_id = source_id)
+                jp2_id = metadata['id']
+            except KeyError:
+                raise KeyError("The values used for observatory, instrument, detector, measurement "
+                               "do not correspond to a source_id. Please check the list using "
+                               "HelioviewerClient.data_sources.")
+
+        URL = ('https://api.helioviewer.org/v2/getJP2Header/?id=' + (jp2_id))
+
+        params = {
+            "action": "getJP2Header",
+            "id" : jp2_id,
+        }
```

Comment:
What does this do?

Reply:
To get the header from the API we need to pass a unique JP2 image identifier which is the jp2_id

Reply:
params is not used anywhere.